extensions/Push/lib/Connector/Webhook.pm

```diff
@@ -94,17 +96,35 @@ sub send {
     $payload->{webhook_name} = $webhook->name;
     $payload->{webhook_id}   = $webhook->id;
 
-    my $bug_data   = $self->_get_bug_data($payload);
-    my $is_private = $bug_data->{is_private};
-    if ($is_private){
-      delete @{$payload}{bug};
-      if($payload->{event}->{action} eq 'modify'){
+    my $bug_data       = $self->_get_bug_data($payload);
+    my $bug_is_private = $bug_data->{is_private};
+    my $target         = $payload->{event}->{target};
+
+    my $target_is_private = $payload->{$target}->{is_private};
+    if ($target_is_private && ($target eq 'attachment' || $target eq 'comment')){
+      my $target_id = $payload->{$target}->{id};
+      delete @{$payload}{$target};
+      $payload->{$target}->{id}         = $target_id;
+      $payload->{$target}->{is_private} = $target_is_private;
+      $payload->{$target}->{bug}        = $bug_data;
+    }
+
+    if ($bug_is_private){
+      if ($target eq 'bug'){
+        delete @{$payload}{bug};
+        $payload->{bug}->{id}         = $bug_data->{id};
+        $payload->{bug}->{is_private} = $bug_is_private;
+      }
+      else{
+        delete @{$payload->{$target}}{bug};
+        $payload->{$target}->{bug}->{id}         = $bug_data->{id};
+        $payload->{$target}->{bug}->{is_private} = $bug_is_private;
```

Comment:
Same here. _int($bug_data->{id}) and _boolean($bug_is_private). Basically anywhere that data is being added to the JSON that is to be sent as part of the payload, needs to use the proper methods from `extensions/lib/Push/lib/Serialize.pm`.